.github/workflows/release.yaml

```diff
@@ -0,0 +1,104 @@
+name: Release on merge to main
+on:
+  push:
+    branches:
+      - main
+
+env:
+  BOT_NAME: newrelic-coreint-bot
+  BOT_EMAIL: coreint-dev@newrelic.com
+
+permissions:
+  contents: write
+
+jobs:
+  test-release-needed:
+    name: Test if release is needed
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      # Validation and generation.
+      - name: Validate that the markdown is correct
+        uses: ./validate-markdown
+      - name: Generate YAML
+        uses: ./generate-yaml
+        with:
+          excluded-dirs: .github
+
+      # Check that a release is needed.
+      - name: Check if the release is empty
+        id: empty
+        uses: ./is-empty
+      - name: Check if the release is held
+        id: held
+        uses: ./is-held
+      - name: Output if the release should be skipped
+        id: output
+        shell: bash
+        run: |
+          echo "skip=${{ steps.empty.outputs.is-empty == 'true' || steps.held.outputs.is-held == 'true' }}" >> $GITHUB_OUTPUT
+    outputs:
+      make-release: ${{ steps.output.outputs.skip }}
+
+  make-release:
+    name: Test if release is needed
+    runs-on: ubuntu-latest
+    needs: [ test-release-needed ]
+    if: ${{ needs.test-release-needed.outputs.make-release }}
+    steps:
+      - name: Generate YAML
+        uses: ./generate-yaml
+        with:
+          excluded-dirs: .github
+
+      # Put release toolkit to work.
+      - name: Link dependencies
+        uses: ./link-dependencies
+      - name: Calculate next version
+        id: version
+        uses: ./next-version
+
+      # Prepare to commit the Changelog.
+      - name: Configure Git
+        run: |
+          git config user.name '${{ env.BOT_NAME }}'
+          git config user.email '${{ env.BOT_EMAIL }}'
+
+      # Create changelog and commit it.
+      - name: Update the markdown
+        uses: ./update-markdown
+        with:
+          version: ${{ steps.version.outputs.next-version }}
+      - name: Commit updated changelog
+        run: |
+          git add CHANGELOG.md
+          git commit -m "Update changelog with changes from ${{ steps.version.outputs.next-version }}"
+          git push -u origin ${{ github.event.repository.default_branch }}
+
+      # Create the release from the commit above using only the changelog for this release.
+      - name: Render the changelog snippet
+        uses: ./render
+        with:
+          version: ${{ steps.version.outputs.next-version }}
+      - name: Create release
+        run: |
+          gh release create \
+            ${{ steps.version.outputs.next-version }} \
+            --title ${{ steps.version.outputs.next-version }} \
+            --target $(git rev-parse HEAD) \
+            --notes-file CHANGELOG.partial.md
+
```

Comment:
Do we need to also update the _major tag_? Example: previous release is `v1.0.3` and next release is `1.1.0`, when `1.1.0` is created, shouldn't the `v1` tag be updated as well?

Reply:
🤔 We may need https://github.com/newrelic/release-toolkit/pull/188 to do it?

Reply:
We could added that feature in a different PR but as #188 was already merged I added that in this PR at [#9d1efb4](https://github.com/newrelic/release-toolkit/pull/187/commits/9d1efb4235bc94dd6c488869950cd7ef41d56678)